src/lib/cms/components/CMSProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import * as Immutable from 'immutable';
+import {IRequestResponse, ICmsDataDbModel} from '../models';
+import {updateDraftCmsData} from '../client';
+
+interface ICMSProviderProps {
+  cmsData: ICmsDataDbModel
+}
+
+interface ICMSProviderState {
+  cmsData: ICmsDataDbModel
+}
+
+export class CMSProvider extends React.Component<ICMSProviderState, ICMSProviderState> {
+
+  static childContextTypes = {
+    cmsData        : React.PropTypes.object.isRequired,
+    onCmsDataUpdate: React.PropTypes.func.isRequired
+  }
+
+  public constructor(props: ICMSProviderState, context?: any) {
+    super(props, context);
+    this.state = {
+      cmsData: this.props.cmsData
+    }
+    this.handleCmsDataUpdate = this.handleCmsDataUpdate.bind(this);
+  }
+
+  public getChildContext(): any {
+    return {
+      cmsData        : this.state.cmsData,
+      onCmsDataUpdate: this.handleCmsDataUpdate
+    };
+  }
+
+  /**
+   * Send request to server to update the CMS data
+   * @param data - CMS data that has been updated
+   * @param path - Path where the data to be updated should be saved
+   */
+  public handleCmsDataUpdate(data: any, path: string[]): void {
+    updateDraftCmsData(data, path)
```

Comment:
Good idea to be sending the partial data to update .. this way you wont have to be re-sending the entire cmsData to the server with every little change